staging/src/k8s.io/client-go/rest/with_retry.go

```diff
@@ -372,6 +379,63 @@ func readAndCloseResponseBody(resp *http.Response) {
 	}
 }
 
+type ResponseCloser struct {
+	once     sync.Once
+	resp     *http.Response
+	respCopy io.ReadWriteCloser
```

Comment:
rather than storing respCopy as a member variable and counting on callers to remember to use it instead of resp.Body, what if we replace resp.Body inside DrainAndClose()?